test-data/callx.yaml

```diff
@@ -0,0 +1,266 @@
+      "r8.type=number", "r8.svalue=1", "r8.uvalue=1",
+      "s[504...511].type=number"]
+
+code:
+  <start>: |
+    callx r8; bpf_map_lookup_elem
+
+post:
+  - r0.type=shared
+  - r0.shared_offset=0
+  - r0.shared_region_size=4
+  - r0.svalue=[0, 2147418112]
+  - r0.uvalue=[0, 2147418112]
+  - r0.svalue=r0.uvalue
+  - r8.type=number
+  - r8.svalue=1
+  - r8.uvalue=1
+  - s[504...511].type=number
+---
+test-case: callx 0
+
+pre: ["r8.type=number", "r8.svalue=1000", "r8.uvalue=1000"]
+
+code:
+  <start>: |
+    callx r8
+
+post:
+  - r8.type=number
+  - r8.svalue=1000
+  - r8.uvalue=1000
+
+messages:
+  - "0: invalid helper function id 1000 (r8.type is helper)"
+---
+test-case: callx with invalid id
+
+pre: ["r8.type=number", "r8.svalue=1000", "r8.uvalue=1000"]
+
+code:
+  <start>: |
+    callx r8
+
+post:
+  - r8.type=number
+  - r8.svalue=1000
+  - r8.uvalue=1000
+
+messages:
+  - "0: invalid helper function id 1000 (r8.type is helper)"
+---
+test-case: callx with interval
+
+pre: ["r8.type=number", "r8.svalue=[1, 2]", "r8.uvalue=[1, 2]"]
+
+code:
+  <start>: |
+    callx r8; bpf_map_lookup_elem
+
+post:
+  - r8.type=number
+  - r8.svalue=[1, 2]
+  - r8.uvalue=[1, 2]
+
+messages:
+  - "0: callx helper function id is not a valid singleton (r8.type is helper)"
+---
+test-case: callx bpf_map_lookup_elem with non-numeric stack key
+
+pre: ["r1.type=map_fd", "r1.map_fd=1",
+      "r2.type=stack",
+      "r8.type=number", "r8.svalue=1", "r8.uvalue=1"]
+
+code:
+  <start>: |
+    callx r8; bpf_map_lookup_elem
+
+post:
+  - r0.type=shared
+  - r0.shared_offset=0
+  - r0.shared_region_size=4
+  - r0.svalue=[0, 2147418112]
+  - r0.uvalue=[0, 2147418112]
+  - r0.svalue=r0.uvalue
+  - r8.type=number
+  - r8.svalue=1
+  - r8.uvalue=1
+
+messages:
+  - "0: Illegal map update with a non-numerical value [-oo-oo) (within stack(r2:key_size(r1)))"
+---
+test-case: callx bpf_map_update_elem with non-numeric stack value
+
+pre: ["r1.type=map_fd", "r1.map_fd=1",
+      "r2.type=stack", "r2.stack_offset=504",
+      "r3.type=stack", "r3.stack_offset=496",
+      "r4.type=number",
+      "r8.type=number", "r8.svalue=2", "r8.uvalue=2",
+      "s[504...511].type=number"]
+
+code:
+  <start>: |
+    callx r8; bpf_map_update_elem
+
+post:
+  - r0.type=number
+  - s[504...511].type=number
+  - r8.type=number
+  - r8.svalue=2
+  - r8.uvalue=2
+
+messages:
+  - "0: Illegal map update with a non-numerical value [496-500) (within stack(r3:value_size(r1)))"
+---
+test-case: callx bpf_map_update_elem with context value
+
+pre: ["r1.type=map_fd", "r1.map_fd=1",
+      "r2.type=stack", "r2.stack_offset=504",
+      "r3.type=ctx", "r3.ctx_offset=0",
+      "r4.type=number",
+      "r8.type=number", "r8.svalue=2", "r8.uvalue=2",
+      "s[504...511].type=number"]
+
+code:
+  <start>: |
+    callx r8; bpf_map_update_elem
+
+post:
+  - r0.type=number
+  - s[504...511].type=number
+  - r8.type=number
+  - r8.svalue=2
+  - r8.uvalue=2
+
+messages:
+  - "0:  (r3.type in {stack, packet})"
```

Comment:
Why double space here? It may be confusing.

Reply:
It was already this way before this PR, but I have now updated ebpf_domain.cpp to have a message, and updated other yaml files that had double space.